armi/reactor/blueprints/blockBlueprint.py

```diff
@@ -240,11 +239,6 @@ def construct(
         b.verifyBlockDims()
         b.spatialGrid = spatialGrid
 
-        if b.spatialGrid is None and cs[globalSettings.CONF_BLOCK_AUTO_GRID]:
```

Comment:
This is the single place this is used inside the framework. And likely anywhere based on shared experience and discussion with some other devs. My understanding is very few if any people set this setting to be false.

We should do one of these things

1. Include the `CONF_BLOCK_AUTO_GRID` setting check inside `HexBlock.autoCreateSpatialGrids`.
2. Include the `CON_BLOCK_AUTO_GRID` setting check inside `HexAssembly.orientBlocks` when prior to calling the auto grid creation
3. Document in release notes and in the `CONF_BLOCK_AUTO_GRID` setting definition that this is removed and should not be used anymore

I'm inclined for option 3 thought it is, technically, API breaking. Someone out there who relies on `CONF_BLOCK_AUTO_GRID = False` will have a problem. My gut is it's a small group, but I've been wrong before and will be wrong again

Reply:
I'm afraid (1) and (2) aren't possible without making changes to like 12 full methods to pass this information around.

My plan was to do (3) and remove this unused setting after this PR, as I was trying to reduce the phase space of this PR, to make it easier to review.

BUT, since you asked for it, sure I can remove the setting in this PR. I honestly don't understand what `False` ever meant in ARMI anyway. As far as I can tell, `False` has not been used in the last 8 years, at least inside TP.

Reply:
done